Bottom: e1d96a3ae963f6ee6de4725161091b05012768a0
Top:    f8ec05d47f9a8e26204964880f0c6c1e60f4187b
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-14 01:18:51 +0100

Refresh of gtk-gl-refactor-gl-colour-setu

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 409c182..6d97de6 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -53,7 +53,7 @@ typedef struct hid_gc_struct
 {
   HID *me_pointer;
 
-  gchar *colorname;
+  const char *colorname;
   gint width;
   gint cap, join;
   gchar xor;
@@ -363,8 +363,6 @@ set_gl_color_for_gc (hidGC gc)
     return;
 
   free (priv->current_colorname);
-  priv->current_colorname = NULL;
-
   priv->current_colorname = strdup (gc->colorname);
 
   if (gport->colormap == NULL)
@@ -453,11 +451,7 @@ set_gl_color_for_gc (hidGC gc)
 void
 ghid_set_color (hidGC gc, const char *name)
 {
-  current_gc = gc;
-
-  gc->colorname = (char *) name;
-
-  set_gl_color_for_gc (gc);
+  gc->colorname = name;
 }
 
 void
